### DISCUSSION TITLE
Finding length of decoded stream?

### DISCUSSION THREAD

**Discussion by olegkudrin** [librasn/rasn#282] [Q&A] [answered]
Suppose there is a tail in the decoding buffer. Is there a way to find out the number of bytes consumed for decode?

**Comment by XAMPPRocky** [comment, marked as answer]
Thank you for your question! You can if you use the `Decoder` structs directly access the `decoded_len` to get the length consumed from the decode. https://docs.rs/rasn/latest/rasn/ber/de/struct.Decoder.html#method.decoded_len

**Comment by olegkudrin** [comment]
Everything is working, thank you.
